ledger/participant-integration-api/src/main/scala/platform/apiserver/RateLimitingInterceptor.scala

```diff
@@ -75,6 +91,34 @@ private[apiserver] final class RateLimitingInterceptor(metrics: Metrics, config:
     }
   }
 
+  private def memoryOverloaded(fullMethodName: String): Either[String, Unit] = {
+    tenuredMemoryPool.fold[Either[String, Unit]](Right(())) { p =>
```

Comment:
To increase robustness of the solution I would suggest dealing explicitly with 2 possible future error cases:
* we encounter a GC that doesn't have any tenured pool
* we encounter a GC with more than one tenured pool
In both cases the mechanism should be disabled and an error message should be printed into the log

In the successful case, I would print a message saying what kind of memory pool we are hooking to and what kind of thresholds we are observing.